To analyze whether the defective percentage varies by center, we can perform a chi-square test for independence using the provided data. This test will help us determine if there is a significant association between the center and the defective status.

In [18]:
import pandas as pd
import scipy.stats as stats

# Load the data from the CSV file
df = pd.read_csv("Costomer+OrderForm.csv")

# Display the loaded data
print(df.head())

  Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free


In [19]:
print(df['Phillippines'].value_counts(),df['Indonesia'].value_counts(),df['Malta'].value_counts(),df['India'].value_counts())

Phillippines
Error Free    271
Defective      29
Name: count, dtype: int64 Indonesia
Error Free    267
Defective      33
Name: count, dtype: int64 Malta
Error Free    269
Defective      31
Name: count, dtype: int64 India
Error Free    280
Defective      20
Name: count, dtype: int64


In [20]:
observed_values=([[271,267,269,280],[29,33,31,20]])
observed_values

[[271, 267, 269, 280], [29, 33, 31, 20]]

In [21]:
chi2_stat, p_value, _, _ = stats.chi2_contingency(observed_values)

# Print results
print(f'Chi-square Statistic: {chi2_stat}')
print(f'P-value: {p_value}')

# Check significance at 5% level
alpha = 0.05
if p_value < alpha:
    print('Reject the null hypothesis. The defective percentage varies by center.')
else:
    print('Fail to reject the null hypothesis. There is no significant variation in the defective percentage by center.')

Chi-square Statistic: 3.858960685820355
P-value: 0.2771020991233135
Fail to reject the null hypothesis. There is no significant variation in the defective percentage by center.


The P-value of the Chi-square test is 0.277

Since the p-value from the Chi-square test is more than the significance level, We can say that we fail to reject the null hypothesis.

Hence, There is no significant variation in the defective percentage by center.